CHANGELOG.md

```diff
@@ -2,49 +2,42 @@
 
 All notable changes to this project will be documented in this file, in reverse chronological order by release.
 
-## 2.7.0 - TBD
+## 3.0.0 - TBD
 
 ### Added
 
-- Nothing.
+- Check for circular references in role hierarchy when using `Role::addChild()`
+  and `Role::addParent()` functions.
 
 ### Changed
 
-- Nothing.
+- `Role::addChild(RoleInterface $child)`, accepts only `RoleInterface` parameter,
+   no string anymore.
+- `Role::addParent(RoleInterface $parent)`, accepts only `RoleInterface`
+   parameter, no string anymore.
+- `Zend\Permissions\Rbac\AssertionInterface`, added the optional parameters
+  `$permission` and `$role` in the `assert()` function, as follows:
+  `assert(Rbac $rbac, string $permission = null, RoleInterface $role = null)`
```

Comment:
Is the permission more likely than the role to be passed? I would have assumed this would be in the following order:

```php
assert(Rbac $rbac, RoleInterface $role = null, string $permission = null)
```

For that matter: are the permission and role arguments actually optional? Will there ever be a case where they are not provided?

Reply:
@weierophinney I added `$role` and `$permission` as optional parameters because in this way we can limit BC breaks from the previous implementation. The actual interface is:
```php
namespace Zend\Permissions\Rbac;
interface AssertionInterface
{
    /**
     * Assertion method - must return a boolean.
     *
     * @param  Rbac    $rbac
     * @return bool
     */
    public function assert(Rbac $rbac);
}
```

Reply:
@weierophinney I changed the order of params, using the follows:

```php
assert(Rbac $rbac, RoleInterface $role = null, string $permission = null)
```